index.js

```diff
@@ -32,6 +32,38 @@ function getMainArgs() {
   return process.argv.slice(2);
 }
 
+function setOptionValue(parseOptions, option, value, result) {
+  const multiple = parseOptions.multiples &&
+    parseOptions.multiples.includes(option);
+  const withValue = parseOptions.withValue &&
+    parseOptions.withValue.includes(option);
+
+  // Normal flag: !withValue && value === undefined
+  // Normal value, withValue && value !== undefined
+  // Special case: withValue && value === undefined
+  //    store as normal for withValue with value undefined
+  // Special case: !withValue && value !== undefined
+  //    store as normal for withValue (and not a flag)
```

Comment:
The developer would like a flag (due to the `!withValue` config) but the user has set a `value` instead.

Should this case be like a truty flag?

```suggestion
  //    store as normal for value (and not a flag)
```

Reply:
The developer may want a flag or may simply have not supplied any configuration. The user has specified a value, like `--foo=bar`. My reasoning is the user has not supplied a flag, they have supplied an option with a value.

(In the `strict` case we could return an error.)

Related issue is #23